### ISSUE TITLE
Adding KDE's KTouch typing tutor

### ISSUE THREAD

**Issue by probitlabs** [NixOS/nixpkgs#36984]
#  `ktouch` package

## Description
It seems like there is no `ktouch` package receipt [existing](https://nixos.org/nixos/packages.html#ktouch), even through its part of the generated [sources file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/kde/srcs.nix) and it seems like there has been a KDE4 package.

Is there an interest in maintaining a package for KDE's typing tutor?

## Reproduction
```fish
[I] ⋊> ~ sudo nix-env -i ktouch                                                                                                                                                                                                                   
error: selector 'ktouch' matches no derivations
```

**Comment by aanderse** [comment]
@probitlabs the ktouch package is under kdeApplications, so:
`nix-env -iA nixos.kdeApplications.ktouch`

Please close this issue.

**Comment by probitlabs** [comment]
@aanderse Thanks for your hint.
Meanwhile, I can see the package on the web: https://nixos.org/nixos/packages.html#ktouch
Oddly, I cannot detect the packages' existence on terminal:

```
[I] # nix-channel --list
nixos https://nixos.org/channels/nixos-unstable

[I] # nix-channel --update; nix-env -qaP ktouch; nix-env -qaP konsole
unpacking channels...
error: selector 'ktouch' matches no derivations
nixos.kdeApplications.konsole  konsole-18.12.0

[I] # nix-env -iA nixos.kdeApplications.ktouch
error: attribute 'ktouch' in selection path 'nixos.kdeApplications.ktouch' not found
```

Why does that contradiction occur?

**Comment by aanderse** [comment]
@probitlabs hmm that is weird. I was able to run the program by running ktouch after dropping into a shell with it via `nix run nixpkgs.kdeApplications.ktouch`, though.

**Comment by probitlabs** [comment]
> [...] I was able to run the program by running ktouch after dropping into a shell with it via `nix run nixpkgs.kdeApplications.ktouch`, though.

Strangely, I am unable.

```
[I] $ nix run nixpkgs.kdeApplications.ktouch
error: attribute 'ktouch' in selection path 'nixpkgs.kdeApplications.ktouch' not found
```

**Comment by aanderse** [comment]
Which version of nix, nixos, etc...?

**Comment by probitlabs** [comment]
```
[I] $ nix --version; nixos-version; nixos-version --revision
nix (Nix) 2.2pre6600_85488a93
19.03pre168613.ffc604e5579 (Koi)
ffc604e55790dbf575187738e1c3778231e8f5a9
```

**Comment by aanderse** [comment]
If you install via `environment.systemPackages = [ pkgs.kdeApplications.ktouch ];` does it work?

**Comment by probitlabs** [comment]
No.

```
[I] # nixos-rebuild switch --show-trace
[...]
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:263:40:
attribute 'ktouch' missing, at /etc/nixos/configuration.nix:65:9
```

**Comment by tilpner** [comment]
What value does your `NIX_PATH` have, and what entries does your `~/.nix-defexpr` have (paste `tree` output)?

**Comment by probitlabs** [comment]
```
[I] # echo $NIX_PATH; tree -L 2 ~/.nix-defexpr/ ~/.nix-defexpr/channels/ ~/.nix-defexpr/channels/nixos/
/root/.nix-defexpr/channels nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixos nixos-config=/etc/nixos/configuration.nix /nix/var/nix/profiles/per-user/root/channels

/root/.nix-defexpr/
└── channels -> /nix/var/nix/profiles/per-user/root/channels

/root/.nix-defexpr/channels/
├── binary-caches -> /nix/store/6vdzq3phgc5hbqqqnid8hpi2vgr6gq70-nixos-19.03pre168613.ffc604e5579/binary-caches
├── manifest.nix -> /nix/store/vimjcw15vp860qj81aspx43h95js23sz-env-manifest.nix
└── nixos -> /nix/store/6vdzq3phgc5hbqqqnid8hpi2vgr6gq70-nixos-19.03pre168613.ffc604e5579/nixos

/root/.nix-defexpr/channels/nixos/
├── COPYING
├── default.nix
├── doc
│   ├── coding-conventions.xml
│   ├── configuration.xml
│   ├── contributing.xml
│   ├── cross-compilation.xml
│   ├── default.nix
│   ├── functions
│   ├── functions.xml
│   ├── introduction.chapter.md
│   ├── languages-frameworks
│   ├── lib-function-docs.nix
│   ├── lib-function-locations.nix
│   ├── Makefile
│   ├── manual.xml
│   ├── meta.xml
│   ├── multiple-output.xml
│   ├── old
│   ├── overlays.xml
│   ├── overrides.css
│   ├── package-notes.xml
│   ├── platform-notes.xml
│   ├── quick-start.xml
│   ├── release-notes.xml
│   ├── reviewing-contributions.xml
│   ├── shell.nix
│   ├── stdenv.xml
│   ├── style.css
│   └── submitting-changes.xml
├── lib
│   ├── asserts.nix
│   ├── attrsets.nix
│   ├── customisation.nix
│   ├── debug.nix
│   ├── default.nix
│   ├── deprecated.nix
│   ├── fetchers.nix
│   ├── filesystem.nix
│   ├── fixed-points.nix
│   ├── generators.nix
│   ├── kernel.nix
│   ├── licenses.nix
│   ├── lists.nix
│   ├── meta.nix
│   ├── minver.nix
│   ├── modules.nix
│   ├── options.nix
│   ├── sources.nix
│   ├── strings.nix
│   ├── strings-with-deps.nix
│   ├── systems
│   ├── tests
│   ├── trivial.nix
│   ├── types.nix
│   ├── versions.nix
│   └── zip-int-bits.nix
├── maintainers
│   ├── maintainer-list.nix
│   └── scripts
├── nixos
│   ├── COPYING
│   ├── default.nix
│   ├── doc
│   ├── lib
│   ├── maintainers
│   ├── modules
│   ├── README
│   ├── release-combined.nix
│   ├── release.nix
│   ├── release-small.nix
│   └── tests
├── nixpkgs -> .
├── pkgs
│   ├── applications
│   ├── build-support
│   ├── common-updater
│   ├── data
│   ├── desktops
│   ├── development
│   ├── games
│   ├── misc
│   ├── os-specific
│   ├── servers
│   ├── shells
│   ├── stdenv
│   ├── test
│   ├── tools
│   └── top-level
├── programs.sqlite
├── README.md
└── svn-revision

35 directories, 61 files
```

**Comment by tilpner** [comment]
Apparently (@samueldr found this), ktouch was added in #53859 and backported to 18.09 #53887. So now ktouch is in staging and release-18.09, but not yet master and thus not unstable either. Weird situation. You can wait until staging reaches unstable, or pick that package from 18.09 for now.

**Comment by probitlabs** [comment]
@tilpner Thank you for clearing things up. Picking the package from stable works:

```
[I] # nix-channel --add https://nixos.org/channels/nixos-18.09 stable; and nix-channel --list
nixos https://nixos.org/channels/nixos-unstable
stable https://nixos.org/channels/nixos-18.09
[I] # nix-channel --update stable
unpacking channels...
created 5 symlinks in user environment
[I] # nix-env -qaP .\*ktouch.\*
stable.kdeApplications.ktouch  ktouch-18.08.0
```

Took some time to fetch some dependencies again on install, but the executable starts.

**Comment by aanderse** [comment]
@probitlabs good to close?

**Comment by probitlabs** [comment]
@aanderse Yes, all fine. However, in the medium term I would be glad to see the the package on unstable too.

**Comment by samueldr** [comment]
It should be available at the latest before the 19.03 branch-off later this month, but hopefully a bit sooner.
